src/main/java/slimeknights/tconstruct/gadgets/entity/shuriken/ShurikenEntityBase.java

```diff
@@ -0,0 +1,62 @@
+package slimeknights.tconstruct.gadgets.entity.shuriken;
+
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.EntityType;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.entity.projectile.ProjectileItemEntity;
+import net.minecraft.network.IPacket;
+import net.minecraft.network.PacketBuffer;
+import net.minecraft.util.DamageSource;
+import net.minecraft.util.math.EntityRayTraceResult;
+import net.minecraft.util.math.vector.Vector3d;
+import net.minecraft.world.World;
+import net.minecraftforge.fml.common.registry.IEntityAdditionalSpawnData;
+import net.minecraftforge.fml.network.NetworkHooks;
+
+import javax.annotation.Nonnull;
+
+public abstract class ShurikenEntityBase extends ProjectileItemEntity implements IEntityAdditionalSpawnData {
+
+  public ShurikenEntityBase(EntityType<? extends ShurikenEntityBase> type, World worldIn) {
+    super(type, worldIn);
+  }
+
+  public ShurikenEntityBase(EntityType<? extends ShurikenEntityBase> type, double x, double y, double z, World worldIn) {
+    super(type, x, y, z, worldIn);
+  }
+
+  public ShurikenEntityBase(EntityType<? extends ShurikenEntityBase> type, LivingEntity livingEntityIn, World worldIn) {
+    super(type, livingEntityIn, worldIn);
+  }
+
+  public abstract float getDamage();
```

Comment:
I tend to like javadoc comments on public methods. Getters are a bit obvious, but I'd rather be consistent than make exceptions. Its a bit less obvious for knockback, helps to be clear the return value is in blocks or alike

Reply:
Done! I don't know exactly what the value is in blocks, but I know some typical values that Minecraft used, so I'll try to get some useful info in there.